model/src/main/java/org/projectnessie/model/Branch.java

```diff
@@ -47,4 +55,26 @@ default void checkName() {
   static Branch of(String name, String hash) {
     return builder().name(name).hash(hash).build();
   }
+
+  /**
+   * Returns a map of metadata properties that describe this branch. Note that these properties
+   * <b>can be added</b> by the server when a {@link Branch} instance is returned. A returned {@link
+   * Branch} instance can then have the following keys:
+   *
+   * <ul>
+   *   <li>{@value Branch#NUM_COMMITS_AHEAD}: number of commits ahead of the default branch
+   *   <li>{@value Branch#NUM_COMMITS_BEHIND}: number of commits behind the default branch
+   *   <li>{@value Branch#HEAD_COMMIT_META}: the commit metadata of the HEAD commit
```

Comment:
Should we specify metadata value types, perhaps?

Reply:
that's the part I'm not sure about